clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10655,6 +10655,8 @@ def err_second_argument_to_cwsc_not_pointer : Error<
 
 def err_vector_incorrect_num_elements : Error<
   "%select{too many|too few}0 elements in vector %select{initialization|operand}3 (expected %1 elements, have %2)">;
+def err_invalid_even_odd_vector_element_count : Error<
+  "invalid element count of %0 in vector %select{initialization|operand}4 (expected an %select{even|odd}3 element count in the range of %1 and %2)">;
```

Comment:
This diagnostic seems odd. You only use it for requiring an even element count, but I'm not actually sure that's the most useful diagnostic for this case.

Really we need a vector whose total size is a multiple of 64-bits.

Reply:
"a vector whose total size is a multiple of 64-bits" could also mean `uint16_t4` ;)
But with a check for 32-bits in the integer arguments as a separate error message, it would be sufficient.

Or perhaps, do you think the AddUint64 function should be generalized to accept any vector of unsigned integers whose total size is a multiple of 64-bits? Like `uint16_t4`, or potentially `uint32_t` vectors longer than 4 in the future?